### PR TITLE
EggNOG: Implement possibility of two stage mode

### PR DESCRIPTION
**Pull request by bernt-matthias**
As described here https://github.com/eggnogdb/eggnog-mapper/wiki/eggNOG-mapper-v2.1.5-to-v2.1.8#Setting_up_large_annotation_jobs

Wondering if I should split this into two tools (maybe also keeping the 'monolithic' one). Advantage would be that admins can set different destinations for the CPU intense search and IO intense annotation stage. 

Also adds the `cache` mode. And makes the tests more specific by using regexes instead of `sim_size`

### CODE REVIEW

**Comment by bernt-matthias** [comment]
Ping @abretaud @jj-umn and @bgruening: what do you think about splitting the tool?

**Comment by bernt-matthias** [comment]
> @bernt-matthias do you have performance problems?
> 
> I guess two tools make technical sense but is this really a problem for our users? How much in UX do we decrease if we split it up? For hard-code people, we can now run this tool twice correct? I guess this is enough to address both use-cases.

One of  my users has a FASTA with 12,000,000 sequences. The search phase runs several days (with good CPU usage on 10cores) and the annotation phase runs more than a week using only a few percent of the 10 cores (currently my max run time). Might be that the new `--scratch_dir` and `--temp_dir` parameters help. I'm happy to test first.

File: tools/eggnog_mapper/eggnog_macros.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
         <version_command>emapper.py --version</version_command>
     </xml>
     <xml name="data_manager_params">
-        <param name="test" type="hidden" value="false" />
+        <param nae="test" type="hidden" value="false" />
```
[the remark this answers is not in the document]
Comment (bernt-matthias) [reply tools/eggnog_mapper/eggnog_macros.xml]: fixes

File: tools/eggnog_mapper/eggnog_macros.xml [hunk 2]
```diff
@@ -64,10 +64,82 @@ python '${__tool_directory__}/data_manager_eggnog.py' --config_file '$out_file'
...
+                </param>
+            </when>
+        </conditional>
+    </xml>
+    <!-- Common options for search filtering (applies to diamond and mmseqs only) -->
+    <xml name="common_search_options">
+        <param argument="--query_cover" type="integer" optional="true" min="0" max="100" label="Minimum query coverage" help="Report only alignments above the given percentage of query cover" />
+        <param argument="--subject_cover" type="integer" optional="true" min="0" max="100" label="Minimum subject coverage" help="Report only alignments above the given percentage of subject cover" />
+        <param argument="--pident" type="integer" optional="true" min="0" max="100" label="Minimum query coverage" help="Report only alignments above the given percentage of identity" />
+        <param argument="--evalue" type="float" optional="true" min="0" label="Minimum query coverage" help=" Report only alignments below or equal the e-value" />
```
Comment (abretaud) [inline tools/eggnog_mapper/eggnog_macros.xml:105]:
```suggestion
        <param argument="--evalue" type="float" optional="true" min="0" label="Minimum query coverage" help="Report only alignments below or equal the e-value" />
```

**Review by abretaud — COMMENTED** [review]
Just a minor typo, the rest looks good to me!
I see the point of splitting the tool, but I fear it would make it a bit too complex for normal users. If you do it, I'd vote for keeping the monolithic one as well

**Review by bgruening — APPROVED** [review]

**Pull request merged**